src/app/BufferedReadCallback.cpp

```diff
@@ -36,13 +36,8 @@ void BufferedReadCallback::OnReportBegin()
 
 void BufferedReadCallback::OnReportEnd()
 {
-    CHIP_ERROR err = DispatchBufferedData(mBufferedPath, StatusIB(), true);
-    if (err != CHIP_NO_ERROR)
-    {
-        mCallback.OnError(err);
-        return;
-    }
-
+    mLastError = DispatchBufferedData(mBufferedPath, StatusIB(), true);
+    ReturnOnFailure(mLastError);
```

Comment:
OK, so this skips calling OnReportEnd and removes the OnError call.  But API guarantees either OnReportEnd or OnError, no?  I don't understand how this is supposed to work.

That said, in the _subscription_ case OnError means something quite specific which does not match what this code ends up doing if we did call OnError.

So what is the actual behavior we are trying for here?

Reply:
sorry, I forgot to push latest change, could you re-review?

Reply:
Again: "what is the actual behavior we are trying for here?"  For reads, and for subscriptions (which may want different behaviors, because OnError has very different meanings for them).